packages/react-components/react-positioning/src/devtools/createPositioningDevTools.ts

```diff
@@ -0,0 +1,82 @@
+import { ReferenceElement } from '@floating-ui/dom';
+import { Container, ContainerData } from './types';
+import { assertContainer, isContainer } from './methods';
+import { POSITIONING_DEV_TOOLS } from './constants';
+
+export type PositioningDevTools = {
+  getContainer(): HTMLElement;
+  getTarget(): ReferenceElement;
+  getData(): ContainerData;
+  getScrollParentByIndex(index: number): HTMLElement;
+  getOverflowBoundaryByIndex(index: number): HTMLElement;
+  getFlipBoundaryByIndex(index: number): HTMLElement;
+  isRef(element?: HTMLElement | null): boolean;
+  withdraw(): void;
+  /**
+   * sets the current inspected element.
+   * 1. if the element is a floating container a reference to that element will be preserved.
+   * 2. if the element is a valid reference then return container's data
+   * 3. if the element is not a valid reference then withdraw reference and return null
+   *
+   * @param element element that was selected
+   */
+  select(element?: HTMLElement | null): ContainerData | null;
+};
+
+export function createPositioningDevTools(): PositioningDevTools {
```

Comment:
why can't this be done on the devtools side apart from types and utilities that aren't used in the production code?